### PR TITLE
[Small Feature] Grid Simplification with Outlier Removal

### PR DESCRIPTION
**Pull request by sgiraudot**
## Rationale

The algorithm `grid_simplify_point_set()` uses a 3D grid of user-defined size to simplify a point cloud by only retaining 1 point per occupied cell.

By adding a minimal number of points in a cell such that a point is retained, this algorithm can also filter out low density areas and outliers: in the case of densely sampled point cloud, this gives better results than using grid simplification _then_ outlier removal, while being very vast (as the grid simplification only relies on an `unordered_set` or `unordered_map`).

## Summary of API changes

A new optional named parameter `min_points_per_cell` is added to `grid_simplify_point_set()`. The default value is `1` to keep the current behavior as default.

## License and copyright ownership

(no change)

## CHANGES.md

TODO

## Submission

* [__User Manual__](https://cgal.github.io/5698/v1/Point_set_processing_3/index.html#title37)
* [__Reference Manual__](https://cgal.github.io/5698/v1/Point_set_processing_3/group__PkgPointSetProcessing3Algorithms.html#gac7c81cc8a2986e3972e86612e4f847a1)

## Status

* Developed and locally tested (GNU/Linux)
* [Small Feature](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Grid_Simplification_with_Outlier_Removal)

### CODE REVIEW

File: Point_set_processing_3/include/CGAL/grid_simplify_point_set.h [hunk 1]
```diff
@@ -222,32 +258,25 @@ grid_simplify_point_set(
   const std::function<bool(double)>& callback = choose_parameter(get_parameter(np, internal_np::callback),
```
Comment (sloriot) [inline Point_set_processing_3/include/CGAL/grid_simplify_point_set.h:258]: New warning:
`/mnt/testsuite/include/CGAL/grid_simplify_point_set.h:258:38: warning: unused variable 'callback' [-Wunused-variable]`

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.5-Ic-1

**Comment by lrineau** [comment]
This small feature has been merged, but I only saw afterwards that the `CHANGES.md` was left unchanged.

**Comment by sloriot** [comment]
@afabri could you please take care of the update?

**Pull request merged**